### PR TITLE
Refactor accusations

### PR DESCRIPTION
**Pull request by keirongulrajani**
## Scope

-  Divided accusations into standalone contracts concentrating on specific accusation types
- Introduced `salt-type` parameter for immutable factory, allowing for a defined contract type to be included in the salt calculation
- includes test rewrite of accusations tests into typescript

## Why?

[MP-590](https://madhive.atlassian.net/browse/MP-590)
[MP-551](https://madhive.atlassian.net/browse/MP-551)

## Todos
If any, what are the follow-up tasks required other than merging this PR?  Have they been arranged?
- [ ] ???

### CODE REVIEW

File: bridge/test/accusations/invalid-transaction-consumption-accusations.test.ts [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+import {
+  addValidators,
+  getAccusationDataForNonExistentUTXOWithInvalidSigGroup,
+  getInvalidAccusationDataWithSpendingValidDeposit,
+  getValidAccusationDataForNonExistentUTXO,
+  getValidAccusationDataForNonExistentUTXOChainId2,
+  getValidAccusationDataForNonExistentUTXOWithInvalidHeight,
+} from "./accusations-test-helpers";
+
+describe("StakingPositionDescriptor: Tests StakingPositionDescriptor methods", async () => {
```
Comment (ricardopinto) [inline bridge/test/accusations/invalid-transaction-consumption-accusations.test.ts:13]: not sure if this is related to staking or accusations

File: bridge/test/accusations/multiple-proposal-accusations.test.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import {
+  addValidators,
+  generateSigAndPClaims0,
+  generateSigAndPClaims1,
+  generateSigAndPClaimsDifferentChainId,
+  generateSigAndPClaimsDifferentHeight,
+  generateSigAndPClaimsDifferentRound,
+} from "./accusations-test-helpers";
+
+describe("StakingPositionDescriptor: Tests StakingPositionDescriptor methods", async () => {
```
Comment (ricardopinto) [inline bridge/test/accusations/multiple-proposal-accusations.test.ts:13]: not sure if this is related to staking or accusations

File: bridge/test/accusations/accusations-library.test.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { assert } from "chai";
+import { ethers } from "hardhat";
+import { AccusationsLibraryMock } from "../../typechain-types";
+import {
+  deployLibrary,
+  generateSigAndPClaims0,
+  generateSigAndPClaims1,
+} from "./accusations-test-helpers";
+
+describe("StakingPositionDescriptor: Tests StakingPositionDescriptor methods", async () => {
```
Comment (ricardopinto) [inline bridge/test/accusations/accusations-library.test.ts:10]: not sure if this is related to staking or accusations

**Review by ricardopinto — CHANGES_REQUESTED** [review]

File: bridge/test/accusations/multiple-proposal-accusations.test.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    accusation = fixture.multipleProposalAccusation;
+  });
+
+  describe("Accusation", async () => {
+    const expectedAddress = "0xe17aB3698B3e93bCE3303Fd5b2E5dB8E376E1cb9";
+    it("deploys to expected address", async function () {
+      const address = accusation.address;
+
+      assert.equal(address, expectedAddress);
```
Comment (ricardopinto) [inline bridge/test/accusations/multiple-proposal-accusations.test.ts:29]: seems like this test is not passing

**Review by ricardopinto — CHANGES_REQUESTED** [review]

File: bridge/test/accusations/invalid-transaction-consumption-accusations.test.ts (hunk 1 above)
Comment (keirongulrajani) [reply bridge/test/accusations/invalid-transaction-consumption-accusations.test.ts]: whoops - copy pasted the boilerplate but forgot to change the names here! Fixed now for this and the other tests too

File: bridge/test/accusations/invalid-transaction-consumption-accusations.test.ts [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+    accusation = fixture.invalidTxConsumptionAccusation;
+  });
+
+  describe("Accusation", async () => {
+    const expectedAddress = "0xCB734f084f21b77D09773Bc07F82A2C39144B952";
+    it("deploys to expected address", async function () {
+      const address = accusation.address;
+
+      assert.equal(address, expectedAddress);
```
Comment (ricardopinto) [inline bridge/test/accusations/invalid-transaction-consumption-accusations.test.ts:29]: this one is not passing in the pipelines

File: bridge/test/accusations/accusations-test-helpers.ts [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+      expect(await validatorPool.isValidator(validator)).to.equal(false);
+      await validatorPool.registerValidators([validator], [0]);
+      expect(await validatorPool.isValidator(validator)).to.equal(true);
+    }
+  }
+};
+
+export const setChainId = async (
+  snapshots: SnapshotsMock | Snapshots,
+  newchainId: number
```
Comment (ricardopinto) [inline bridge/test/accusations/accusations-test-helpers.ts:504]: `newchainId` is defined but never used

**Review by ricardopinto — CHANGES_REQUESTED** [review]
we can merge this to main as soon as pipeline is green

**Review by ricardopinto — APPROVED** [review]

**Pull request merged**